src/app/content/hooks/receivePageNotFoundId.ts

```diff
@@ -16,6 +21,9 @@ export const receivePageNotFoundIdHookBody: ActionHookBody<typeof receivePageNot
       return;
     }
   }
+  if (bookConfig?.retired) {
+    return Promise.reject(new BookNotFoundError(`Could not resolve uuid: ${book?.id}`));
+  }
```

Comment:
@TomWoodward this seems to work for pages of retired books where no redirect is specified. There's a bit of a delay though so you see the old book for a moment.